src/nvim/syntax.c

```diff
@@ -7685,6 +7689,10 @@ void highlight_changed(void)
     }
   }
   highlight_ga.ga_len = hlcnt;
+
+  ui_call_highlight_info_set(changed_highlights);
```

Comment:
@bfredl would removing the *highlight_info_set and call use mode_info_set if needed (on a cursor hl change) be ok ?

Reply:
Yes, the only issue is that we might need some kind of batching. If the user reloads their color scheme it should only be sent once in total. Might follow similar scheme as `pending_cursor_update` in `ui.c`: when a highlight group changes set it to true, and in `ui_cursor_shape` send the `mode_info_set` if the flag is true.